packages/@aws-cdk/toolkit/lib/actions/deploy.ts

```diff
@@ -0,0 +1,228 @@
+import { StackSelector } from '../types';
+
+export type DeploymentMethod = DirectDeploymentMethod | ChangeSetDeploymentMethod;
+
+export interface DirectDeploymentMethod {
+  /**
+   * Use stack APIs to the deploy stack changes
+   */
+  readonly method: 'direct';
+}
+
+export interface ChangeSetDeploymentMethod {
+  /**
+   * Use change-set APIS to deploy a stack changes
+   */
+  readonly method: 'change-set';
+
+  /**
+   * Whether to execute the changeset or leave it in review.
+   *
+   * @default true
+   */
+  readonly execute?: boolean;
+
+  /**
+   * Optional name to use for the CloudFormation change set.
+   * If not provided, a name will be generated automatically.
+   */
+  readonly changeSetName?: string;
```

Comment:
i feel like we should have a linter rule that makes sure optional properties have `@default` on them